Core/TimerInterface.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  TimerInterface.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
```

Comment:
Made this interface so I can mock the Timer in the tests for [AnimatableTypingTextModel](https://github.com/duckduckgo/iOS/pull/3015/files#diff-5c6803ef20a5e1c9404650e676a1d3a9271430bda06f8a575172f9982a7e13d7)